src/app/clusters/groups-server/groups-server.cpp

```diff
@@ -232,141 +261,137 @@ bool emberAfGroupsClusterViewGroupCallback(app::CommandHandler * commandObj, con
     return true;
 }
 
-bool emberAfGroupsClusterGetGroupMembershipCallback(app::CommandHandler * commandObj, const app::ConcreteCommandPath & commandPath,
-                                                    const Commands::GetGroupMembership::DecodableType & commandData)
+struct GetGroupMembershipResponse
 {
-    auto & groupCount = commandData.groupCount;
-    auto & groupList  = commandData.groupList;
-
-    EmberStatus status = EMBER_ZCL_STATUS_FAILURE;
-    uint8_t i, j;
-    uint8_t count = 0;
-    uint8_t list[EMBER_BINDING_TABLE_SIZE << 1];
-    uint8_t listLen = 0;
-    CHIP_ERROR err  = CHIP_NO_ERROR;
-
-    emberAfGroupsClusterPrint("RX: GetGroupMembership [");
-
-    // For all networks, Get Group Membership commands may be sent either
-    // unicast or broadcast (removing the ZLL-specific limitation to unicast).
-
-    // When Group Count is zero, respond with a list of all active groups.
-    // Otherwise, respond with a list of matches.
-    // TODO: https://github.com/project-chip/connectedhomeip/issues/10335
-    if (groupCount == 0)
+public:
+    // Use GetCommandId instead of commandId directly to avoid naming conflict with CommandIdentification in ExecutionOfACommand
+    static constexpr CommandId GetCommandId() { return Commands::GetGroupMembershipResponse::Id; }
+    static constexpr ClusterId GetClusterId() { return Groups::Id; }
+
+    GetGroupMembershipResponse(const Commands::GetGroupMembership::DecodableType & data,
+                               GroupDataProvider::GroupMappingIterator * iter) :
+        mCommandData(data),
+        mIterator(iter)
+    {}
+
+    const Commands::GetGroupMembership::DecodableType & mCommandData;
+    GroupDataProvider::GroupMappingIterator * mIterator = nullptr;
+    // A capacity of 0xFF means that it is unknown if any further groups may be
+    // added.  Each group requires a binding and, because the binding table is
+    // used for other purposes besides groups, we can't be sure what the
+    // capacity will be in the future.
+    uint8_t mCapacity = 0xff;
```

Comment:
This is unused... Move the comment to where the 0xff is actually encoded, perhaps?